vulcanai2/tests/models/test_dnn.py

```diff
@@ -81,8 +129,70 @@ def test_unfreeze_noclass(self, dnn_noclass):
         for params in dnn_noclass.network.parameters():
             assert params.requires_grad is True
 
-    def test_add_input_network(self, dnn_noclass, dnn_class):
-        """Test add input Network functionality."""
-        dnn_class.add_input_network(dnn_noclass)
-        assert dnn_class.input_networks[dnn_noclass.name] is dnn_noclass
-        assert dnn_class.in_dim == dnn_noclass.out_dim
+    def test_fit_multi_input(self, multi_input_dnn_class,
+                             multi_input_dnn_train_loader,
+                             multi_input_dnn_test_loader):
+        """Test for fit function."""
+        init_weights = pickle.loads(pickle.dumps(
+            multi_input_dnn_class.network[0]._kernel.weight.detach()))
+        multi_input_dnn_class_no_fit = pickle.loads(
+            pickle.dumps(multi_input_dnn_class))
+        parameters1 = multi_input_dnn_class_no_fit.parameters()
+        try:
+            multi_input_dnn_class.fit(
+                multi_input_dnn_train_loader,
+                multi_input_dnn_test_loader,
+                2)
+        except RuntimeError:
+            logger.error("The network multi_input_dnn_class failed to train.")
+        finally:
+            parameters2 = multi_input_dnn_class.parameters()
+            trained_weights = multi_input_dnn_class.network[0]._kernel.weight.detach()
+            
+            # Sanity check if the network parameters are training
+            assert not (torch.equal(init_weights.cpu(), trained_weights.cpu()))
```

Comment:
does this work with float values?

Reply:
yeah the weights are float values.

Reply:
@priyatharsan @rfratila  Yeah I know they're float values. My point is that generally equality functions don't work with float values; you have to check that they're approximately equal or equal up to a certain number of decimal points. Looking at the implementation of the function https://github.com/torch/torch7/pull/470 I'm not sure that it actually fulfills what we need... it may just be checking the size, and thus this condition will always be True.

Reply:
@nirtiac yes this is a good point. @joemehltretter uses an approximate comparison with bounds for testing the selu weight inits that we can use here. Could you adapt some of that code and apply it here?

Reply:
Or use np.testing.assert_almost_equal(res, target_res)

Reply:
yeah I think that's what he uses